Formula/infer.rb

```diff
@@ -122,13 +89,34 @@ def install
         if (s != NULL) {
           *s = 42;
         }
-
         return 0;
       }
     EOS
 
-    shell_output("#{bin}/infer --fail-on-issue -P -- clang -c FailingTest.c", 2)
-    shell_output("#{bin}/infer --fail-on-issue -P -- clang -c PassingTest.c")
+    no_issues_output = "\n  No issues found  \n"
+
+    failing_c_output = <<-EOS
+
+FailingTest.c:5: error: Null Dereference
+  pointer `s` last assigned on line 4 could be null and is dereferenced at line 5, column 3.
+  3. int main() {
+  4.   int *s = NULL;
+  5.   *s = 42;
+       ^
+  6.   return 0;
+  7. }
+
+
+Found 1 issue
+          Issue Type(ISSUED_TYPE_ID): #
+  Null Dereference(NULL_DEREFERENCE): 1
+    EOS
```

Comment:
```suggestion
    failing_c_output = <<~EOS

      FailingTest.c:5: error: Null Dereference
        pointer `s` last assigned on line 4 could be null and is dereferenced at line 5, column 3.
        3. int main() {
        4.   int *s = NULL;
        5.   *s = 42;
             ^
        6.   return 0;
        7. }


      Found 1 issue
                Issue Type(ISSUED_TYPE_ID): #
        Null Dereference(NULL_DEREFERENCE): 1
    EOS
```
Doesn't this work?

Reply:
It doesn't appear to:

```
 FailingTest.c:5: error: Null Dereference
-pointer `s` last assigned on line 4 could be null and is dereferenced at line 5, column 3.
-3. int main() {
-4.   int *s = NULL;
-5.   *s = 42;
-     ^
-6.   return 0;
-7. }
+  pointer `s` last assigned on line 4 could be null and is dereferenced at line 5, column 3.
+  3. int main() {
+  4.   int *s = NULL;
+  5.   *s = 42;
+       ^
+  6.   return 0;
+  7. }


 Found 1 issue
-        Issue Type(ISSUED_TYPE_ID): #
-Null Dereference(NULL_DEREFERENCE): 1
+          Issue Type(ISSUED_TYPE_ID): #
+  Null Dereference(NULL_DEREFERENCE): 1
 "
```

I've tried a few combinations here and none of them seem to work 😕

Reply:
Use `<<-EOS` everywhere. That should work unless you need trailing whitespace.

Reply:
To be clear, the one I tried and pasted here was using `<<~EOS` with *six* leading spaces

Reply:
```suggestion
    failing_c_output = <<~EOS

      FailingTest.c:5: error: Null Dereference
      \ \ pointer `s` last assigned on line 4 could be null and is dereferenced at line 5, column 3.
      \ \ 3. int main() {
      \ \ 4.   int *s = NULL;
      \ \ 5.   *s = 42;
      \ \      ^
      \ \ 6.   return 0;
      \ \ 7. }


      Found 1 issue
      \ \       Issue Type(ISSUED_TYPE_ID): #
      \ \ Null Dereference(NULL_DEREFERENCE): 1
    EOS
```
Not sure why it's eating all of your leading spaces (it shouldn't be eating spaces after `FailingTest.c`), but you can always escape them.

Reply:
Actually, you probably just need to escape the first space.

Reply:
```diff
-
-FailingTest.c:5: error: Null Dereference
-  pointer `s` last assigned on line 4 could be null and is dereferenced at line 5, column 3.
-  3. int main() {
-  4.   int *s = NULL;
-  5.   *s = 42;
-       ^
-  6.   return 0;
-  7. }
-
-
-Found 1 issue
-          Issue Type(ISSUED_TYPE_ID): #
-  Null Dereference(NULL_DEREFERENCE): 1
+
+      FailingTest.c:5: error: Null Dereference
+      \  pointer `s` last assigned on line 4 could be null and is dereferenced at line 5, column 3.
+      \  3. int main() {
+      \  4.   int *s = NULL;
+      \  5.   *s = 42;
+      \       ^
+      \  6.   return 0;
+      \  7. }
+
+
+      Found 1 issue
+      \          Issue Type(ISSUED_TYPE_ID): #
+      \  Null Dereference(NULL_DEREFERENCE): 1
```

Gives:

```
Error: infer: failed
An exception occurred within a child process:
  Minitest::Assertion: --- expected
+++ actual
@@ -1,15 +1,15 @@
-"
-      FailingTest.c:5: error: Null Dereference
-        pointer `s` last assigned on line 4 could be null and is dereferenced at line 5, column 3.
-        3. int main() {
-        4.   int *s = NULL;
-        5.   *s = 42;
-             ^
-        6.   return 0;
-        7. }
-
-
-      Found 1 issue
-                Issue Type(ISSUED_TYPE_ID): #
-        Null Dereference(NULL_DEREFERENCE): 1
+"
+FailingTest.c:5: error: Null Dereference
+  pointer `s` last assigned on line 4 could be null and is dereferenced at line 5, column 3.
+  3. int main() {
+  4.   int *s = NULL;
+  5.   *s = 42;
+       ^
+  6.   return 0;
+  7. }
+
+
+Found 1 issue
+          Issue Type(ISSUED_TYPE_ID): #
+  Null Dereference(NULL_DEREFERENCE): 1
 "
```

Sorry if it seems that I am being hopeless here, I'm really trying to get this "happy", but it seems something isn't playing ball ...

Reply:
We got there 🥳 

```bash
1 file inspected, no offenses detected
```

Reply:
Thanks for sticking with me, @carlocab 💪